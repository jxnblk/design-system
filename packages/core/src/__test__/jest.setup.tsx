import React from 'react'
import { ThemeProvider } from 'styled-components'

import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'
import renderer from 'react-test-renderer'

import { createTheme } from '..'

global.rendererCreateWithTheme = (tree) =>
  renderer.create(<ThemeProvider theme={createTheme()}>{tree}</ThemeProvider>)

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
