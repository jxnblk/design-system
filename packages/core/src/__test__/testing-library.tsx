import { render, RenderResult } from '@testing-library/react'
import PropTypes from 'prop-types'
import React from 'react'

import { ThemeProvider } from '..'

const WithThemeProvider = ({ children }) => <ThemeProvider>{children}</ThemeProvider>
WithThemeProvider.propTypes = {
  children: PropTypes.node,
}

const customRender: (Element, any?) => RenderResult = (ui, options) =>
  render(ui, { wrapper: WithThemeProvider, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }

export const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms))
