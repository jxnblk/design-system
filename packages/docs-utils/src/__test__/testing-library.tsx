import { render, RenderResult } from '@testing-library/react'
import { ThemeProvider } from 'pcln-design-system'
import PropTypes from 'prop-types'
import React from 'react'

const WithThemeProvider = ({ children }) => <ThemeProvider>{children}</ThemeProvider>
WithThemeProvider.propTypes = {
  children: PropTypes.node,
}

// @ts-ignore
const customRender: (Element, any?) => RenderResult = (ui, options) =>
  render(ui, { wrapper: WithThemeProvider, ...options })

// re-export everything
export * from '@testing-library/react'
// override render method
export { customRender as render }
export const sleep = (ms: number) => new Promise((r) => setTimeout(r, ms))
