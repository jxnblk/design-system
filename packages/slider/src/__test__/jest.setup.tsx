import React from 'react'
import { ThemeProvider } from 'pcln-design-system'

import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'
import renderer from 'react-test-renderer'

global.rendererCreateWithTheme = (tree) => renderer.create(<ThemeProvider>{tree}</ThemeProvider>)

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
