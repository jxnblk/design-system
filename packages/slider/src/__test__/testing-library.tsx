import type { RenderOptions, RenderResult } from '@testing-library/react'
import type { ReactElement } from 'react'

import { render } from '@testing-library/react'
import { ThemeProvider } from 'pcln-design-system'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <ThemeProvider>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
