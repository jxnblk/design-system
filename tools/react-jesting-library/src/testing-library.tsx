import React, { FC, ReactElement } from 'react'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import { ThemeProvider } from 'pcln-design-system'

const AllTheProviders: FC = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>
}

/**
 * customRender wraps the render method of \@testing-library/react and wraps it in our
 * custom AllTheProviders which provides a pcln-design-system theme
 *
 * @public
 */
const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
