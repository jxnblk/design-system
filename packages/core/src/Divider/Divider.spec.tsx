import React from 'react'
import { render, screen } from '../__test__/testing-library'
import { Divider, theme } from '..'

describe('Divider', () => {
  test('renders', () => {
    const json = rendererCreateWithTheme(<Divider />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('width prop sets width', () => {
    const json = rendererCreateWithTheme(<Divider width={1 / 2} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('width', '50%')
  })

  test('m prop sets margin', () => {
    const json = rendererCreateWithTheme(<Divider m={2} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('margin', theme.space[2])
  })

  test('borderColor prop sets borderColor', () => {
    const json = rendererCreateWithTheme(<Divider borderColor='blue' />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('border-color', theme.colors.blue)
  })

  test('override default margin left and right', () => {
    render(<Divider mx={2} data-testid='divider' />)

    expect(screen.getByTestId('divider')).toHaveStyleRule('margin-left', theme.space[2])
    expect(screen.getByTestId('divider')).toHaveStyleRule('margin-right', theme.space[2])
  })
})
