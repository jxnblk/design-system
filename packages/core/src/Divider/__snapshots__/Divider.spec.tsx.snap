// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider borderColor prop sets borderColor 1`] = `
.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  border: 0;
  border-bottom-style: solid;
  border-bottom-width: 1px;
  border-color: #0068ef;
  background-color: #0068ef;
  margin-left: 0px;
  margin-right: 0px;
}

<div
  className="c0"
>
  <hr
    className="c1"
    color="border"
  />
</div>
`;

exports[`Divider m prop sets margin 1`] = `
.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  border: 0;
  border-bottom-style: solid;
  border-bottom-width: 1px;
  border-color: #c0cad5;
  background-color: #c0cad5;
  margin: 8px;
  margin-left: 0px;
  margin-right: 0px;
}

<div
  className="c0"
>
  <hr
    className="c1"
    color="border"
  />
</div>
`;

exports[`Divider renders 1`] = `
.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  border: 0;
  border-bottom-style: solid;
  border-bottom-width: 1px;
  border-color: #c0cad5;
  background-color: #c0cad5;
  margin-left: 0px;
  margin-right: 0px;
}

<div
  className="c0"
>
  <hr
    className="c1"
    color="border"
  />
</div>
`;

exports[`Divider width prop sets width 1`] = `
.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  border: 0;
  border-bottom-style: solid;
  border-bottom-width: 1px;
  border-color: #c0cad5;
  background-color: #c0cad5;
  width: 50%;
  margin-left: 0px;
  margin-right: 0px;
}

<div
  className="c0"
>
  <hr
    className="c1"
    color="border"
    width={0.5}
  />
</div>
`;
